models/yang/sonic/sonic-nat.yang

```diff
@@ -0,0 +1,218 @@
+module sonic-nat {
+    namespace "http://github.com/Azure/sonic-nat";
+    prefix snat;
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-common {
+        prefix scommon;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC NAT";
+
+    revision 2019-10-30 {
+        description
+            "Initial revision.";
+    }
+
+
+    container sonic-nat {
+
+        container STATIC_NAPT {
+
+            list STATIC_NAPT_LIST {
+                key "global_ip ip_protocol global_l4_port";
+
+                leaf global_ip {
+                    type inet:ipv4-address;
+                }
+
+                leaf ip_protocol {
+                    type uint8 {
+                        range "1|2|6|17|46|47|51|103|115";
+                    }
+                }
+
+                leaf global_l4_port {
+                    type uint16;
+                }
+
+                leaf local_ip {
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf local_port {
+                    mandatory true;
+                    type uint16;
+                }
+
+                leaf nat_type {
+                    type enumeration {
+                        enum SNAT;
+                        enum DNAT;
+                    }
+                    default DNAT;
+                }
+
+                leaf twice_nat_id {
+                    type uint16 {
+                        range "1..9999" {
+                            error-message "Invalid Group Id for twice NAPT.";
+                        }
+                    }
+                }
+            }
+        }
+
+        container STATIC_NAT {
+            list STATIC_NAT_LIST {
+                key "global_ip";
+
+                leaf global_ip {
+                    type inet:ipv4-address;
+                }
+
+                leaf local_ip {
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf nat_type {
+                    type enumeration {
+                        enum SNAT;
+                        enum DNAT;
+                    }
+                    default DNAT;
+                }
+
+                leaf twice_nat_id {
```

Comment:
Avoid same data definition in multiple places. Try to use grouping for similar objects.

Reply:
Addressed